### PR TITLE
Expose LedgerDb API for looking up txs by nonce

### PR DESCRIPTION
**Pull request by eranrund**
### Motivation

As I work through integrating the new transaction types into SCP I realized I needed ledger APIs for checking if MintTx and SetMintConfigTx nonces appear in the ledger. The change proposed here provides a similar API to the KeyImage check we already have.

I changed the LMDB databases from `nonce` -> `Serialized tx object containing the nonce` to `nonce` -> `block index`. This would allow us to provide information to callers about which block the tx appeared in. If looking up the tx by nonce is needed, that can be done by fetching the `BlockContents` for the given block index, so no functionality is lost.

### In this PR
* Two new public LedgerDb APIs needed for validation/scp integration
* Changing internal LMDB databases to better suit our usage patterns

### CODE REVIEW

File: ledger/db/src/mint_config_store.rs [hunk 1]
```diff
@@ -139,8 +141,38 @@ impl MintConfigStore {
...
+            // MintConfigs -> ActiveMintConfigs
+            let active_mint_configs = ActiveMintConfigs::from(set_mint_config_tx);
+
+            // Store in database
+            db_transaction.put(
+                self.block_index_by_set_mint_config_tx_nonce,
+                &set_mint_config_tx.prefix.nonce,
+                &block_index_bytes,
+                WriteFlags::NO_OVERWRITE, /* this ensures we do not overwrite a nonce that was
+                                           * already used */
```
Comment (remoun) [inline ledger/db/src/mint_config_store.rs:165]: Nit: Switch to line comment(s) above the value.
```suggestion
                //  ensures we do not overwrite a nonce that was already used
                WriteFlags::NO_OVERWRITE,
```

**Review by remoun — APPROVED** [review]

**Review by garbageslam — APPROVED** [review]
LGTM

File: ledger/db/src/ledger_trait.rs [hunk 2]
```diff
@@ -95,4 +95,14 @@ pub trait Ledger: Send {
     /// Returns an empty array of no mint configurations are active for the
```
Comment (mfaulk) [inline ledger/db/src/ledger_trait.rs:95]: of --> if

**Review by mfaulk — APPROVED** [review]

**Review by jcape — APPROVED** [review]
LGTM, but one thing that would help later is renaming `SetMintConfigTx` to just `MintConfigTx`. That would preclude lexical contortions like `get_set_mint_config_tx()` ;-)

**Comment by eranrund** [comment]
> LGTM, but one thing that would help later is renaming `SetMintConfigTx` to just `MintConfigTx`. That would preclude lexical contortions like `get_set_mint_config_tx()` ;-)

Good idea, I'll do that after the 3 PRs are merged.

**Pull request merged**